internal/store/engine_state.go

```diff
@@ -82,6 +82,7 @@ type EngineState struct {
 	WatchSettings model.WatchSettings
 
 	PendingConfigFileChanges map[string]time.Time
+	PendingTiltfileTrigger   TriggerTiltfileAction
```

Comment:
Hmmmm...TiltfileState should have a BuildReason on it already. have you considered using that field?

we've been generally trying to unify these data structures, rather than inventing new ones for Tiltfile

Reply:
ah i didn't realize that field existed, ty.